datafusion/expr/src/logical_plan/builder.rs

```diff
@@ -145,7 +150,7 @@ impl LogicalPlanBuilder {
             coerce_plan_expr_for_schema(&recursive_term, self.plan.schema())?;
         Ok(Self::from(LogicalPlan::RecursiveQuery(RecursiveQuery {
             name,
-            static_term: Arc::new(self.plan.clone()),
+            static_term: Arc::clone(&self.plan),
```

Comment:
This seems the only place a deep clone is "saved"?

Reply:
Another example in line 470.

```
                project((*input).clone(), expr)
                project(unwrap_arc(input), expr)
```

Reply:
Ah yes, here it might avoid the clone as well :+1:

Reply:
FWIW I think we can avoid the clone by changing the signature from

```rust
    pub fn to_recursive_query(
        &self,
```

to 
```rust
    pub fn to_recursive_query(
        self,
```

(the builder should be consumed)

Reply:
> Another example in line 470.

I think since this is an associated function it could also be done indpendently to the change of the internal implementation of `LogicalaPlanBuilder`